tests/test_bookmarks_import.py

```
@@ -0,0 +1,210 @@
+
+
+# --- Encoding -------------------------------------------------------------
+#
+# The shipped eve_bookmark_helper.ini is UTF-16 LE with a BOM, because that
+# is what AutoHotkey's IniWrite produces on a Unicode build. Reading it as
+# UTF-8 leaves a NUL after every character, so every section header fails
+# _parse_ini's "]" test and the entire file imports as nothing -- which the
+# caller then saved over the user's real settings while reporting success.
+# These pin the decode, and the shape of the bug, so it cannot come back.
+
+def _encoded(text, encoding):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_encoded`

(ANN202)